### DISCUSSION TITLE
Static pages and dynamic at the same time

### DISCUSSION THREAD

**Discussion by kcubero27** [vercel/next.js#15702] [Help] [answered]
Hi 👋 

I'm currently building a website where I can static and dynamic pages. The landing page is static but one of the other pages is completely dynamic and needs getServerSideProps. Therefore, I ended up removing next export from my built command. I was hoping if there is any option to use next export and pages with getServerSideProps with at the same time. 

The command I execute is `next build && next export -o build` but it fails because of the getServerSideProps function:

```
> using build directory: /Users/kcubero/Projects/gitlab.com/web-v2/.next
  copying "static build" directory
> No "exportPathMap" found in "next.config.js". Generating map from "./pages"
  launching 15 workers
  copying "public" directory
[==  ] Exporting (0/4)
Error occurred prerendering page "/s/[search]". Read more: https://err.sh/next.js/prerender-error
Error: Error for page /s/[search]: pages with `getServerSideProps` can not be exported. See more info here: https://err.sh/next.js/gssp-export
    at Object.exportPage [as default] (/Users/kcubero/Projects/gitlab.com/web-v2/node_modules/next/dist/export/worker.js:14:254)
Exporting (4/4)
Error: Export encountered errors on following paths:
	/s/[search]
    at exportApp (/Users/kcubero/Projects/gitlab.com/web-v2/node_modules/next/dist/export/index.js:24:1103)
```

Thanks for your help and patience!

**Comment by jamesmosier** [comment, marked as answer]
Not currently, but this issue and discussion would be good to keep an eye on as they deal with the same question:

https://github.com/vercel/next.js/issues/15674
https://github.com/vercel/next.js/discussions/15440

**Reply by kcubero27** [reply]
Thanks @jamesmosier 
This makes a lot of sense. Hope they do it soon as I think it would be amazing to have both. 
I guess for now I will just do the build and wait for this feature 👍
